common/djangoapps/third_party_auth/admin.py

```diff
@@ -31,8 +40,15 @@ def get_list_display(self, request):
 admin.site.register(OAuth2ProviderConfig, OAuth2ProviderConfigAdmin)
 
 
+class SAMLProviderConfigForm(forms.ModelForm):
+    """ Django Admin form class for SAMLProviderConfig """
+    backend_name = forms.ChoiceField(choices=((name, name) for name in _PSA_SAML_BACKENDS))
+
+
 class SAMLProviderConfigAdmin(KeyedConfigurationModelAdmin):
     """ Django Admin class for SAMLProviderConfig """
+    form = SAMLProviderConfigForm
+
```

Comment:
@bradenmacdonald I've added form and moved choices from models. Does this make sense to you now.? 
Thanks.